doc/api/v8.md

```diff
@@ -946,43 +943,63 @@ const zlib = require('node:zlib');
 const path = require('node:path');
 const assert = require('node:assert');
 
-const {
-  isBuildingSnapshot,
-  addSerializeCallback,
-  addDeserializeCallback,
-  setDeserializeMainFunction,
-} = require('node:v8').startupSnapshot;
+const v8 = require('node:v8');
 
-const filePath = path.resolve(__dirname, '../x1024.txt');
-const storage = {};
+class BookShelf {
+  storage = new Map();
 
-assert(isBuildingSnapshot());
+  // Reading a series of files from directory and store them into storage.
+  constructor(directory, books) {
+    for (const book of books) {
+      this.storage.set(book, fs.readFileSync(path.join(directory, book)));
+    }
+  }
 
-addSerializeCallback(({ filePath }) => {
-  storage[filePath] = zlib.gzipSync(fs.readFileSync(filePath));
-}, { filePath });
+  static compressAll(shelf) {
+    for (const [ book, content ] of shelf.storage) {
+      shelf.storage.set(book, zlib.gzipSync(content));
+    }
+  }
 
-addDeserializeCallback(({ filePath }) => {
-  storage[filePath] = zlib.gunzipSync(storage[filePath]);
-}, { filePath });
+  static decompressAll(shelf) {
+    for (const [ book, content ] of shelf.storage) {
+      shelf.storage.set(book, zlib.gunzipSync(content));
+    }
+  }
+}
 
-setDeserializeMainFunction(({ filePath }) => {
-  console.log(storage[filePath].toString());
-}, { filePath });
+// __dirname here is where the snapshot script is placed
+// during snapshot building time.
+const shelf = new BookShelf(__dirname, [
+  'book1.en_US.txt',
+  'book1.es_ES.txt',
+  'book2.zh_CN.txt',
+]);
+
+assert(v8.startupSnapshot.isBuildingSnapshot());
+// On snapshot serialization, compress the books to reduce size.
+v8.startupSnapshot.addSerializeCallback(BookShelf.compressAll, shelf);
+// On snapshot deserialization, decompress the books.
+v8.startupSnapshot.addDeserializeCallback(BookShelf.decompressAll, shelf);
+v8.startupSnapshot.setDeserializeMainFunction((shelf) => {
+  // process.env and process.argv are refreshed during snapshot
+  // deserialization.
+  const lang = process.env.BOOK_LANG || 'en_US';
+  const book = process.argv[1];
+  const name = `${book}.${lang}.txt`;
+  console.log(shelf.storage.get(name));
+}, shelf);
 ```
 
-The resulted binary will simply print the data deserialized from the snapshot
-during start up:
+The resulted binary will get print the data deserialized from the snapshot
+during start up, using the refreshed `process.env` and `process.argv` of
+the launched process:
 
 ```console
-$ out/Release/node
-# Prints content of ./test/fixtures/x1024.txt
+$ BOOK_LANG=es_ES node --snapshot-blob snapshot.blob book1
+# Prints content of book1.es_ES.txt deserialized from the snapshot.
 ```
 
-Currently the API is only available to a Node.js instance launched from the
```

Comment:
I think this paragraph is still applicable since we can not build a new startup snapshot from a userland snapshot, right?